lib/ansible/modules/system/parted.py

```diff
@@ -372,7 +372,11 @@ def get_unlabeled_device_info(device, unit):
     parted cannot work because of a missing label. It always returns a 'unknown'
     label.
     """
-    device_name = os.path.basename(device)
+    if os.path.islink(device):
+        devicelink = os.readlink(device)
+        device_name = os.path.basename(devicelink)
```

Comment:
thinking about this more, this should be a totally diff path i.e: 
```
'dev': device
'alias': original
```
instead of duplicating the data since we should already be providing the same device data .. or just append an 'alias' to the original devices (caller code might need changes)